### PR TITLE
Better error reporting

### PR DESCRIPTION
**Pull request by aljazerzen**

### CODE REVIEW

File: prql/tests/integration/compile.rs [hunk 1]
```diff
@@ -16,8 +16,8 @@ nodes:
 }
 
 #[test]
-fn transpile_variables() -> Result<()> {
-    assert_snapshot!(compile("select 1")?, @r###"
+fn transpile_variables() {
```
Comment (max-sixty) [inline prql/tests/integration/compile.rs:19]: Do you prefer not to have `Result`s in tests? I rather like them because it saves lots of `.unwrap`s...

**Review by max-sixty — COMMENTED** [review]
Excellent!

File: prql/tests/integration/compile.rs (hunk 1 above)
Comment (aljazerzen) [reply prql/tests/integration/compile.rs:19]: I would have to write `From<(String, Option<SourceLocation>)> for Error` for `?` to work.

That's why I removed results. There is also no ` -> Result<()>` and `Ok(())` for each function, so overall, that may be less characters of code.

Comment (max-sixty) [reply prql/tests/integration/compile.rs:19]: Ah OK, so just for these functions

**Pull request merged**